pkg/rng/entropy_test.go

```diff
@@ -0,0 +1,281 @@
+				testutil.SkipIfInVMTest(t)
+				err = os.Chmod(HwRandomCurrentFile, 0333)
+				if err != nil {
+					t.Error(err)
+				}
+				err = setAvailableTRNG()
+				if err == nil {
+					t.Error(err)
+				}
+				err = os.Chmod(HwRandomCurrentFile, 0755)
+				if err != nil {
+					t.Error(err)
+				}
+			}
+			err = os.RemoveAll(path)
+			if err != nil {
+				t.Error(err)
+			}
+			HwRandomAvailableFile = "/sys/class/misc/hw_random/rng_available"
+			HwRandomCurrentFile = "/sys/class/misc/hw_random/rng_current"
+		})
+	}
+}
+
+func TestUpdateLinuxRandomness(t *testing.T) {
+	for _, tt := range []struct {
+		name       string
+		setup      func(*testing.T)
+		shouldFail bool
+	}{
+		{
+			name: "sufficient randomness",
+			setup: func(t *testing.T) {
+				f, err := os.Create(HwRandomAvailableFile)
+				if err != nil {
+					t.Error(err)
+				}
+				_, err = f.WriteString("tpm-rng")
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+				f, err = os.Create(HwRandomDevice)
+				if err != nil {
+					t.Error(err)
+				}
+				b := [128]byte{}
+				_, err = f.Write(b[:])
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+				f, err = os.Create(RandomDevice)
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+				f, err = os.Create(RandomEntropyAvailableFile)
+				if err != nil {
+					t.Error(err)
+				}
+				_, err = f.WriteString("4000\n")
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+			},
+			shouldFail: false,
+		},
+		{
+			name:       "no trng",
+			setup:      func(t *testing.T) {},
+			shouldFail: true,
+		},
+		{
+			name: "no hwrng",
+			setup: func(t *testing.T) {
+				f, err := os.Create(HwRandomAvailableFile)
+				if err != nil {
+					t.Error(err)
+				}
+				_, err = f.WriteString("tpm-rng")
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+			},
+			shouldFail: true,
+		},
+		{
+			name: "no random",
+			setup: func(t *testing.T) {
+				f, err := os.Create(HwRandomAvailableFile)
+				if err != nil {
+					t.Error(err)
+				}
+				_, err = f.WriteString("tpm-rng")
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+				f, err = os.Create(HwRandomDevice)
+				if err != nil {
+					t.Error(err)
+				}
+				f.Close()
+			},
+			shouldFail: true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			path, err := ioutil.TempDir("", "tmp-*")
+			if err != nil {
+				t.Error(err)
+			}
+			HwRandomAvailableFile = path + "rng_available"
+			HwRandomCurrentFile = path + "rng_current"
+			RandomDevice = path + "random"
+			HwRandomDevice = path + "hwrng"
+			RandomEntropyAvailableFile = path + "entropy_avail"
+			tt.setup(t)
+
+			if !tt.shouldFail {
+				err := UpdateLinuxRandomness()
+				if err != nil {
+					t.Error(err)
+				}
+			} else {
+				err := UpdateLinuxRandomness()
+				if err == nil {
+					t.Error(err)
+				}
+			}
+
+			err = os.RemoveAll(path)
+			if err != nil {
+				t.Error(err)
+			}
+			HwRandomAvailableFile = "/sys/class/misc/hw_random/rng_available"
```

Comment:
This kind of code suggests we also need an interface type for accessing these files. This code is extremely linux-specific.

Reply:
That's not really in the scope of adding coverage but I agree. I renamed the files and added a build constraint to them.